app/lib/utilities.ts

```diff
@@ -0,0 +1,19 @@
+const formatAmountToHumanReadable = (amountStr: string) => {
+  const amount = parseFloat(amountStr)
+  if (amount >= 1000000000) {
+    return (amount / 1000000000).toFixed(2) + 'B'
+  }
+  if (amount >= 1000000) {
+    return (amount / 1000000).toFixed(2) + 'M'
+  }
+  if (amount >= 1000) {
+    return (amount / 1000).toFixed(2) + 'K'
+  }
+  return amount.toFixed(2)
+}
+
+const getAssetCode = (asset: string) => {
```

Comment:
[note]

It seems `assetCode`, is used in the response from horizon.

- `assetCode: 'APSUSDM'`
- `soldAssetCode: 'BTC'`
- `counterAssetCode: 'CRED'`

Reply:
To render an asset that has the "normal" properties use Asset.tsx. See Payment.tsx as an example that uses it. Just pass in the related 3 props for each asset code and it will handle 'native' to 'XLM'.

Handling the codes with ":" in the value was needed for the liquidity pool operations but maybe not this specific case your seeing?  Or are you not seeing ":" at all anymore? could be that was changed in horizon ...  check liquidity pool operations and see ...

Reply:
Here's a response for liquidityPools from horizon regarding (see: `reserves`)
`asset` can be either ...
- `'AUTO:GBI36OIJT3C2P2WN7RLYQFLIAP6RW5QEUKBYONGNTHQTZQBJNOVLDB2O'` or
- `'native'`

The method, `getAssetCode`, is to handle the two cases.

```
// liquidityPools
{
  links: {
    self: {
      href: 'https://horizon.stellar.org/liquidity_pools/fffe949fa72b0957b381229a39a74a11add3a563b1f6a50b05a161ee622d2e81'
    },
    transactions: {
      href: 'https://horizon.stellar.org/liquidity_pools/fffe949fa72b0957b381229a39a74a11add3a563b1f6a50b05a161ee622d2e81/transactions{?cursor,limit,order}',
      templated: true
    },
    operations: {
      href: 'https://horizon.stellar.org/liquidity_pools/fffe949fa72b0957b381229a39a74a11add3a563b1f6a50b05a161ee622d2e81/operations{?cursor,limit,order}',
      templated: true
    }
  },
  id: 'fffe949fa72b0957b381229a39a74a11add3a563b1f6a50b05a161ee622d2e81',
  pagingToken: 'fffe949fa72b0957b381229a39a74a11add3a563b1f6a50b05a161ee622d2e81',
  feeBp: 30,
  type: 'constant_product',
  totalTrustlines: '1',
  totalShares: '53.8429211',
  reserves: [
    {
      asset: 'AUTO:GBI36OIJT3C2P2WN7RLYQFLIAP6RW5QEUKBYONGNTHQTZQBJNOVLDB2O',
      amount: '6187.0784186'
    },
    {
      asset: 'VEND:GBZWXTV66FIXOLM7TINV2H4LD2HSDS6ZK3FANGVK6YCAFBGEN52T7E2I',
      amount: '0.4701031'
    }
  ],
  lastModifiedLedger: 48708267,
  lastModifiedTime: '2023-10-24T23:07:28Z'
}
```

---

Probably my following comment is misleadiing.

> It seems assetCode, is used in the response from horizon.

I just wanted to say I used `assetCode` instead of `assetStr`, which is used in some places in the existing code.
Because I've seen the following responses from horizon.

```
// effects
{
  :
  soldAssetCode: 'BTC',
  :
}
```

```
// operatinos
{
  :
  sellingAssetCode: 'STP',
  :
}
```

Reply:
oh i see, yes a number of keys for assetCodes are different depending on the response. because a lot of responses have asset pairs so 2 assets at least.